### PR TITLE
CMake Boost Option

### PR DESCRIPTION
**Pull request by offa**
Adds `INFLUXCXX_WITH_BOOST` option (default: `ON`), see #80.

@autoantwort: Can you have a look at the PR related to packaging?

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -10,13 +10,13 @@ target_include_directories(InfluxDB-Http SYSTEM PUBLIC $<TARGET_PROPERTY:CURL::l
...
-    $<$<BOOL:${Boost_FOUND}>:UDP.cxx>
-    $<$<BOOL:${Boost_FOUND}>:UnixSocket.cxx>
+    $<$<NOT:$<BOOL:${INFLUXCXX_WITH_BOOST}>>:NoBoostSupport.cxx>
+    $<$<BOOL:${INFLUXCXX_WITH_BOOST}>:BoostSupport.cxx>
+    $<$<BOOL:${INFLUXCXX_WITH_BOOST}>:UDP.cxx>
+    $<$<BOOL:${INFLUXCXX_WITH_BOOST}>:UnixSocket.cxx>
     )
 target_include_directories(InfluxDB-BoostSupport PRIVATE ${INTERNAL_INCLUDE_DIRS})
-target_link_libraries(InfluxDB-BoostSupport PRIVATE $<$<BOOL:${Boost_FOUND}>:Boost::system>)
+target_link_libraries(InfluxDB-BoostSupport PRIVATE $<$<BOOL:${INFLUXCXX_WITH_BOOST}>:Boost::system>)
```
Comment (autoantwort) [inline src/CMakeLists.txt:20]:
```suggestion
target_include_directories(InfluxDB-BoostSupport PRIVATE $<$<BOOL:${INFLUXCXX_WITH_BOOST}>:${Boost_INCLUDE_DIRS}>)
```
Did you not also need something like this to add the paths to the boost headers to the include path?

Comment (offa) [reply src/CMakeLists.txt:20]: Boost system should propagate the paths already, but yeah, better make it more explicit.

**Comment by offa** [comment]
@autoantwort Let me know if I can merge.

**Comment by autoantwort** [comment]
I am currently testing it. Do you also want to release a new version then?

**Comment by offa** [comment]
> I am currently testing it. Do you also want to release a new version then?

Yes, so you can use a release tag.

**Comment by autoantwort** [comment]
Great. LGTM, you can merge

**Pull request merged**